NAudio.Midi/Midi/MidiInterop.cs

```diff
@@ -72,7 +72,7 @@ public enum MidiOutMessage
 
         // http://msdn.microsoft.com/en-us/library/dd798450%28VS.85%29.aspx
         [DllImport("winmm.dll")]
-        public static extern MmResult midiInAddBuffer(IntPtr hMidiIn, ref MIDIHDR lpMidiInHdr, int uSize);
+        public static extern MmResult midiInAddBuffer(IntPtr hMidiIn, IntPtr lpMidiInHdr, int uSize);
```

Comment:
this is technically a breaking change to the NAudio API. Hopefully won't affect many people but will need to mention this in the next release notes

Reply:
It could technically be a breaking change, but the original methods were not used and I could not get it to work with ref MIDIHDR. I have just retried and it crashes out with an uncatchable access violation (0xC0000005 from Win32). So now for future use there is at least some working example code :-)